blueocean-dashboard/src/main/js/components/InputStep.jsx

```diff
@@ -69,17 +70,17 @@ export default class InputStep extends Component {
      * Submit the form as "cancel" out of the state data id.
      */
     cancelForm() {
-        const { id } = this.state;
-        parameterApi.cancelInputParameter(this.state.href, id);
+        const { href, id } = this.state;
+        parameterApi.cancelInputParameter(href, id);
```

Comment:
@tfennelly for me it reduces redundant information while reading the code. 

```
parameterApi.cancelInputParameter(this.state.href, this.state.id);
```

I consider the information `this.state` as disturbing, since I consider `href` and `id` as `const` which I do not care where they come from. It is more I can now do:

```
const { href } = this.props;
const { id } = this.state;
```
 
and 

```
parameterApi.cancelInputParameter(href, id);
```

stays the same.

When reading the diff like this for me it is cristal clear that I have change the `href ` source from `state` to `props`, this is not that obvious when the whole `cancelInputParameter` line will be changed. In that case I need to really compare whether the call has changed or only source .